stock_mts_mto_rule/models/stock_rule.py

```diff
@@ -93,7 +93,7 @@ def _run_split_procurement(self, procurements):
                 # to adjust the product's free qty
                 group_id = mts_procurement.values.get("group_id")
                 group_domain = expression.AND(
-                    [domain, [("group_id", "=", group_id.id)]]
+                    [domain, [("group_id", "=", group_id and group_id.id or False)]]
```

Comment:
Actually, I have a hard time understanding why this block of code with the `moves_to_assign` exists...
The comment is about adapting free qty, which is not used by the module AFAIK.
I don't really like it because I don't think it is the purpose of this module to assign stuff...One may not want to assign stock moves automatically this way. But this matter is not related to the PR...

Anyway, if this part of code is really useful I guess the change make sense. But, without the group it could trigger the assignment of a huge number of moves and it could become a problem (from a performance point of view for instance).

Hard to approve the change since I don't understand this part...

Reply:
Mmmh, long time. I need to check in which case this is needed.

I think this is in case of inter warehouse operations with stock requests (and no procurement group).

Reply:
looks reasonable